routers/web/repo/release_test.go

```diff
@@ -73,6 +74,12 @@ func TestNewReleasesList(t *testing.T) {
 	test.LoadGitRepo(t, ctx)
 	t.Cleanup(func() { ctx.Repo.GitRepo.Close() })
 
+	var err error
+	ctx.Data["NumReleases"], err = repo_model.GetReleaseCountByRepoID(ctx, ctx.Repo.Repository.ID, repo_model.FindReleasesOptions{
+		IncludeDrafts: ctx.Repo.CanWrite(unit_model.TypeReleases),
+	})
+	assert.NoError(t, err)
```

Comment:
Is there an assertion for `ctx.Data["NumReleases"]`?

Reply:
Hmm .... I think the test code is not ideal. Calling `Releases` directly makes the middlewares never executed.

Reply:
Sorry for replying so late. I agree with your opinion that the test code is not ideal. Right now this unit test cannot work without the `RepoAssignment` middleware. I tried moving it to integration tests, then I found there was already a `TestViewReleaseListNoLogin` which tests the release list.

https://github.com/go-gitea/gitea/blob/5600504eed2473691b86962be2f9733e6bfce879/tests/integration/release_test.go#L134-L174

Maybe we can remove `TestNewReleasesList` from unit tests?